src/ring_theory/perfection.lean

```diff
@@ -127,11 +139,11 @@ instance perfect_ring : perfect_ring (ring.perfection R p) p :=
   pth_root_frobenius' := congr_fun $ congr_arg ring_hom.to_fun $ @pth_root_frobenius R _ p _ _ }
 
 instance ring (R : Type u₁) [comm_ring R] [char_p R p] : ring (ring.perfection R p) :=
-((ring.perfection R p).to_subring $ λ n, by simp_rw [← frobenius_def, pi.neg_apply,
+((ring.perfection_subsemiring R p).to_subring $ λ n, by simp_rw [← frobenius_def, pi.neg_apply,
     pi.one_apply, ring_hom.map_neg, ring_hom.map_one]).to_ring
 
 instance comm_ring (R : Type u₁) [comm_ring R] [char_p R p] : comm_ring (ring.perfection R p) :=
-((ring.perfection R p).to_subring $ λ n, by simp_rw [← frobenius_def, pi.neg_apply,
+((ring.perfection_subsemiring R p).to_subring $ λ n, by simp_rw [← frobenius_def, pi.neg_apply,
```

Comment:
This is probably worth extracting to `ring.perfection_subring`

Reply:
Done